core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -17,33 +17,64 @@
 import lombok.Setter;
 import nl.nn.adapterframework.doc.IbisDoc;
 import nl.nn.adapterframework.doc.IbisDocRef;
+import nl.nn.adapterframework.doc.Utils;
 import nl.nn.adapterframework.util.LogUtil;
 
 public class FrankDocModel {
-	private static final String JAVA_STRING = "java.lang.String";
-
 	private static Logger log = LogUtil.getLogger(FrankDocModel.class);
 
 	private @Getter List<FrankDocGroup> groups;
 	private @Getter Map<String, FrankElement> allElements = new HashMap<>();
+	private @Getter Map<String, ElementType> allTypes = new HashMap<>();
 
 	public FrankDocModel() {
 		groups = new ArrayList<>();
 	}
 
-	public FrankElement findOrCreateFrankElement(Class<?> clazz) {
+	public ElementType findOrCreateElementType(Class<?> clazz, ConfigChildDictionary dictionary) {
```

Comment:
Ik stel voor dat ConfigChildDictionary een veld wordt van FrankDocModel, dat je meegeeft bij het aanroepen van de constructor, zodat je zeker weet dat er er voordat je het model gaat maken.
Je moet het niet steeds doorgeven als parameter, want hij verandert nooit.

Reply:
Akkoord. Ik kan dit doen als ik antwoord heb op mijn nieuwe vragen over ConfigChildDictionary.

Reply:
Het model bevat nu een Map<String, ConfigChildSetterDescriptor>. De constructor van het model maakt een lege Map. Die is handig bij het unit testen. Er is een method om deze map te vullen op basis van een (fake) digester-rules.xml file.